src/directory-sync/utils/get-primary-email.ts

```diff
@@ -0,0 +1,6 @@
+import { UserWithGroups } from '../interfaces/user.interface';
+
+export function getPrimaryEmail(user: UserWithGroups): string | undefined {
```

Comment:
We can make this take just a plain `User`.

`UserWithGroups` is a supertype of `User`, so it can also be passed since it structurally satisfies the `User` type.